chainer/functions/activation/softmax.py

```diff
@@ -36,17 +37,17 @@ def forward(self, x):
             one = numpy.array(1, dtype=oz_dtype).ctypes
             zero = numpy.array(0, dtype=oz_dtype).ctypes
             handle = cudnn.get_handle()
-            x_cube = x[0].reshape(x[0].shape[:2] + (-1, 1))
+            x_cube = x[0].reshape(self._get_cube_shape(x[0].shape))
             desc = cudnn.create_tensor_descriptor(x_cube)
             self.y = xp.empty_like(x[0])
             libcudnn.softmaxForward(
                 handle, _algorithm, _mode, one.data, desc.value,
                 x_cube.data.ptr, zero.data, desc.value,
                 self.y.data.ptr)
         else:
```

Comment:
Thank you or summarizing the error conditions. cuDNN version is filtered in the `if` statement at L:35 (`_cudnn_version >= 3000`), so cuDNN is used only when the version >= 3.0 for this softmax operation. It might mean the CUDA version is the cause in the case of the first row result in the table. I'll try to make sure it.